DLL_Project/Detours/JobGiver_GetFood.cs

```diff
@@ -71,13 +71,21 @@ protected override Job TryGiveJob( Pawn pawn )
                 }
                 if( foodSource is Building_AutomatedFactory )
                 {
+                    //Fixed the exepction from passing a Factory into the WillIngestStackCountOf function.
+                    //However, truely fixing this will require detouring FoodUtilities to get proper functionality.
                     var FS = foodSource as Building_AutomatedFactory;
-                    if( foodDef == null )
+                    foodDef = FS.BestProduct(FoodSynthesis.IsMeal, FoodSynthesis.SortMeal);
+                    if( !FS.HasEnoughResourcesInHoppersFor(foodDef) )
                     {
                         //CCL_Log.Message( string.Format( "Hopper for {0} needs filling", foodSource.ThingID ) );
                         hopperNeedsFilling = true;
                         hopper = FS.AdjacentReachableHopper( pawn );
                     }
+                    //TODO I think this is broken for factories. The code does not confirm there's material to fill the hopper available.
+                    //Not an issue with nutrient paste, since if you can't fill a nutrient paste hopper you're probably starving anyway.
+                    //But if you have a factory that makes Lavish Meals from meat. Colonists will wander around starving
+                    //trying to get meat even if there's fine meals in storage. I think anyway.
+                    //Also, current code would ignore a Fine meal factory if the Lavish meal factory has unreachable hoppers.

```

Comment:
BestProduct() looks at the entire list of available recipes and all the resources available.  If it doesn't return a null then the def it returns can indeed be produced and calling HasEnoughResourceInHoppersFor() for the def returned is redundant.  If BestProduct() returns null then the factory either has no recipes or not enough resources available for any recipes it does have.

If BestProduct() returns a def when there isn't the resources to produce it then BestProduct() has a bug which needs to be fixed.